packages/components/src/inputs/src/InputGroup.module.css

```diff
@@ -0,0 +1,92 @@
+.hop-InputGroup  {
+    --input-group-background: var(--hop-neutral-surface);
+    --input-group-border: var(--hop-neutral-border);
+
+    position: relative;
+
+    display: inline-flex;
+    gap: var(--hop-space-inline-sm);
+    align-items: center;
+
+    box-sizing: border-box;
+    inline-size: fit-content;
+    block-size: 2.5rem;
+    padding-inline: var(--hop-space-inset-md) var(--hop-space-inset-md);
+
+    background: var(--input-group-background);
+    border: 0.0625rem solid var(--input-group-border);
+    border-radius: var(--hop-shape-rounded-md)
+}
+
+.hop-InputGroup--fluid,
+.hop-InputGroup--fluid .hop-InputGroup__input {
+    inline-size: 100%
+}
+
+.hop-InputGroup:has(.hop-InputGroup__input[data-focus-within]),
+.hop-InputGroup:has(.hop-InputGroup__input[data-hovered]) {
+    --input-group-background: var(--hop-neutral-surface-hover);
+    --input-group-border: var(--hop-neutral-border-hover)
+}
+
+.hop-InputGroup[data-focus-within]:has(.hop-InputGroup__input[data-focus-visible])  {
+    outline: 0.125rem solid var(--hop-primary-border-focus);
+    outline-offset: 0.125rem;
+}
+
+.hop-InputGroup:has(.hop-InputGroup__input[data-invalid])  {
+    --input-group-border: var(--hop-danger-border-strong)
+}
+
+.hop-InputGroup--sm {
+    block-size: 2rem;
+}
+
+.hop-InputGroup--fluid {
+    inline-size: 100%
+}
+
+.hop-InputGroup__input {
+    flex-grow: 1;
+
+    block-size: 2rem;
+    padding: 0;
+
+    font-family: var(--hop-body-md-font-family);
+    font-size: var(--hop-body-md-font-size);
+    font-weight: var(--hop-body-md-font-weight);
+    line-height: var(--hop-body-md-line-height);
+    color: var(--hop-neutral-text);
```

Comment:
I imagine this will be reset, so it should be a variable like --input-color.
The font properties should probably be component variables at least.